File: csharp/allergies/Allergies.cs
using System;
using System.Linq;

[Flags]
public enum Allergen
{
    Eggs = 1,
    Peanuts = 2,
    Shellfish = 4,
    Strawberries = 8,
    Tomatoes = 16,
    Chocolate = 32,
    Pollen = 64,
    Cats = 128
}

public class Allergies
{
    private readonly Allergen _allergen;
    
    public Allergies(int mask) =>
        _allergen = (Allergen)mask;

    public bool IsAllergicTo(Allergen allergen) =>
        _allergen.HasFlag(allergen);

    public Allergen[] List() =>
        Enum
            .GetValues<Allergen>()
            .Where(IsAllergicTo)
            .ToArray();
}

File: nim/allergies/Iteration2/allergies.nim

import strutils

type
  
  Allergies* = object
    score*: int

  Allergen = enum
    eggs
    peanuts
    shellfish
    strawberries
    tomatoes
    chocolate
    pollen
    cats

  AllergenFlags = set[Allergen]

proc flags(allergies: Allergies): AllergenFlags = 
  cast[AllergenFlags](allergies.score)

proc isAllergicTo*(allergies: Allergies, allergenInput: string): bool =
  let allergen: Allergen = parseEnum[Allergen](allergenInput)
  allergies.flags().contains(allergen)

proc lst*(allergies: Allergies): seq[string] =
  for allergen in allergies.flags():
    result.add($allergen)
